### PR TITLE
scenario checker for algorithm features

### PR DESCRIPTION
**Pull request by damirpolat**
I modified checker tool to also do a simple check for scenarios with algorithm features. Please let me know if I forgot anything.

Thank you, 
Damir

### CODE REVIEW

**Comment by larskotthoff** [comment]
Thanks, looks good to me. Could you please have a look @mlindauer ?

File: data_check_tool_python/src/coseal_reader.py [hunk 1]
```diff
@@ -783,3 +948,115 @@ def remove_features(self):
                     
         return True
         
+    def remove_algorithm_features(self):
+        '''
```
Comment (mlindauer) [inline data_check_tool_python/src/coseal_reader.py:952]: please add a docstring explain what this function is for

File: data_check_tool_python/src/coseal_reader.py [hunk 2]
```diff
@@ -783,3 +948,115 @@ def remove_features(self):
...
+                    total_cost += algo_._feature_group_cost_dict[f_step]
+            for un_step in unused_steps:        # remove step status if unused 
+                del algo_._features_status[un_step]
+            
+            algo_._feature_cost_total = total_cost
+            
+        for un_feature_indx in unused_index_features:
+            self.metainfo.features.pop(un_feature_indx)        
+        
+        #=======================================================================
```
Comment (mlindauer) [inline data_check_tool_python/src/coseal_reader.py:1054]: remove dead code?

**Review by mlindauer — APPROVED** [review]
Thanks for the PR. Only two little comments. Otherwise, I'm happy to have it merged.

File: data_check_tool_python/src/coseal_reader.py (hunk 2 above)
Comment (damirpolat) [reply data_check_tool_python/src/coseal_reader.py]: done

File: data_check_tool_python/src/coseal_reader.py (hunk 1 above)
Comment (damirpolat) [reply data_check_tool_python/src/coseal_reader.py:945]: I updated the docstring

**Comment by larskotthoff** [comment]
Thanks, merging!

**Pull request merged**